### PR TITLE
Block resolving the v2Resolutions promise on the logic actually finishing…

### PR DESCRIPTION
**Pull request by matthew-andrews**
:grin:

### CODE REVIEW

File: server/models/article-v3.js [hunk 1]
```diff
@@ -225,8 +225,7 @@ function mainImageV2(articleV2) {
 }
 
 function v2Resolutions(articleV2) {
-	const toResolve = ['bodyXML', 'openingXML'];
-	toResolve.forEach(item => {
+	return Promise.all(['bodyXML', 'openingXML'].map(item => {
```
Comment (i-like-robots) [inline server/models/article-v3.js]: Should this be filter?

Comment (matthew-andrews) [inline server/models/article-v3.js]: It needs both…

File: server/models/article-v3.js [hunk 2]
```diff
@@ -225,32 +225,29 @@ function mainImageV2(articleV2) {
...
-				articleV2[item] = preresolveImages.replaceImageSets(articleV2[item], imageSets, mediaResources);
-				return articleV2;
-			})
-			.catch(function(error) {
-				logger.warn(`Images could not be resolved: ${error.toString()}`);
-				return articleV2;
-			});
-		}
+	return Promise.all(['bodyXML', 'openingXML']
+		.filter(item => return articleV2[item])
```
Comment (i-like-robots) [inline server/models/article-v3.js]: return?

**Comment by matthew-andrews** [comment]
Oops :-)
https://github.com/Financial-Times/next-es-interface/pull/276/files#r51110447

**Pull request merged**